src/widgets/clicker.rs

```diff
@@ -1,73 +1,179 @@
 use iced::advanced::layout::{self, Layout};
 use iced::advanced::renderer;
-use iced::advanced::widget::{self, Widget};
-use iced::mouse;
+use iced::advanced::widget::{self, Tree, Widget};
+use iced::{advanced::Clipboard, advanced::Shell, touch};
+use iced::{event, mouse, Event};
 use iced::{Color, Element, Length, Rectangle, Size};
 
-pub struct Clicker {
+pub struct Clicker<Message> {
```

Comment:
I can't see where the color RED is set for clicked state.

What about making the color unpressed and pressed as parameters, so the widget can be reused more?

I think I would reserve RED for errors, no?

I think WHITE and BLACK makes more sense for ON / OFF - and still looks quite good?

WDYT?